xtask/src/release.rs

```diff
@@ -68,87 +65,92 @@ impl flags::Release {
-                .into()
-        };
+    /// Get the current version of the crate from the manifest.
+    fn get_version(&self) -> Result<String> {
+        let manifest_toml = read_file(project_root()?.join(&self.name).join("Cargo.toml"))?;
+        let manifest: CargoManifest = from_toml_str(&manifest_toml)?;
 
-        Ok(config)
+        Ok(manifest.package.version)
     }
 }
 
-/// The configuration used to make a release to GitHub.
-#[derive(Debug, Default)]
-struct GithubConfig {
-    /// The local name of the git remote to push to, from the config file.
-    remote: String,
-
-    /// The username to use for authentication, from the config file.
-    user: String,
+/// The required cargo manifest data of a crate.
+#[derive(Debug, Deserialize)]
+struct CargoManifest {
+    /// The package information.
+    package: CargoPackage,
+}
 
-    /// The personal access token to use for authentication, from the config file.
-    token: String,
+/// The required package information from a crate's cargo manifest.
+#[derive(Debug, Deserialize)]
+struct CargoPackage {
+    /// The package version.
+    version: String,
+}
 
-    /// The remote repository on GitHub as `owner/repo`.
-    repo: String,
+/// String manipulations for crate release.
+trait StrExt {
+    /// Remove soft line breaks.
+    fn trim_softbreaks(&self) -> String;
 }
 
-impl GithubConfig {
-    /// Create an empty config.
-    fn new() -> Self {
-        Default::default()
-    }
+impl StrExt for str {
+    fn trim_softbreaks(&self) -> String {
+        let mut string = "".to_owned();
+        let mut current = 0;
+        let mut chars = self.char_indices();
+
+        while let Some(p) = self[current..].find('\n') {
+            let pos = current + p;
+            string.push_str(&self[current..pos]);
+            // Keep hard line breaks (multiple `\n`s).
+            if self[pos..].starts_with("\n\n") {
+                let next = self[pos..].find(|c: char| c != '\n').unwrap_or(0);
+                string.push_str(&self[pos..(pos + next)]);
+                current = pos + next;
+            } else if let Some(p) = self[pos..].find(|c: char| !c.is_ascii_whitespace()) {
+                // Keep line break before list items (`\n` + whitespaces + `*` + whitespaces).
+                // Remove line break and keep one whitespace otherwise.
+                let (_, char) = chars.find(|(i, _)| *i == pos + p).unwrap();
+                if char == '*' {
+                    match chars.next() {
+                        Some((_, next_char)) if next_char.is_ascii_whitespace() => {
+                            string.push('\n');
+                            current = pos + 1;
+                            continue;
+                        }
+                        _ => {}
+                    }
+                }
+                string.push(' ');
+                current = pos + p
+            }
+        }
+
+        string.push_str(&self[(current..self.len())]);
```

Comment:
Instead of `string.push_str(something); string` at the end, you could just return `string + something`.